py/server/tests/test_numba_guvectorize.py

```diff
@@ -90,5 +92,15 @@ def g(x, res):
         self.assertEqual(t.columns[2].data_type, dtypes.long_array)
 
 
+    def test_2d_array_output(self):
+        @guvectorize([(int64[:], int64[:, :])], "(m)->(m, m)", nopython=True)
+        def g(x, res):
+            for i in range(x.shape[0]):
+                res[i] = x[i] + 5
+
+        t = (empty_table(100).update(["X=i%10", "Y=ii"]).group_by("X")
+             .update("Z=(long[][])g(Y)"))
```

Comment:
If this feature is working properly, it seems like the cast should not be needed here.